### PR TITLE
Comment feature

### PR DESCRIPTION
**Pull request by Officer-R**

### CODE REVIEW

File: src/Components/CommentComponent/Comment.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  createCommentHandler = async (e) => {
+    const comment = {
+      comment: this.state.comment,
+      createdAt: new Date().getTime(),
+      postID: this.props.postID,
+    }
+    const commentId = await createComment(this.props.postID, comment)
+    const newComment = {
+      commentId: commentId,
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Why not just `id`?

File: src/Components/CommentComponent/Comment.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          onChange={this.commentInputHandler}
+          value={this.state.comment}
+        />
+        <button
+          style={{ display: "block" }}
+          disabled={!this.state.comment}
+          onClick={this.createCommentHandler}
+        >
+          Submit
+        </button>
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Move comment input to a separate component. Better to always have controlled input as separate components.

File: src/Components/CommentComponent/Comment.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        {this.state.comment_arr && this.state.comment_arr.length
+          ? this.state.comment_arr.map((comment) => {
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Deconstruct comment_arr from state, gives better readability

File: src/Components/CommentComponent/Comment.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              return (
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Return not necessary

File: src/Components/PostsComponent/PostDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from "react"
+import getUniquePost from "../../helpers/getUniquePost"
+import Comment from "../CommentComponent/Comment"
+
+class PostDetails extends Component {
+  state = {
+    uniquePost: [],
```
Comment (digi0ps) [inline src/Components/PostsComponent/PostDetails.jsx]: Why is unique post an empty array?

File: src/Components/PostsComponent/PostDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    loading: true,
+  }
+  async componentDidMount() {
+    const uniquePost = await getUniquePost(this.props.id)
+    this.setState({ uniquePost: uniquePost, loading: false })
```
Comment (digi0ps) [inline src/Components/PostsComponent/PostDetails.jsx]:
```
this.setState({ uniquePost, loading: false })
```
Shorter way

File: src/Components/PostsComponent/PostDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  render() {
+    const post = this.state.uniquePost
+    if (this.state.loading) return "loading..."
+    return (
+      <div>
+        <h1>{post.title}</h1>
+        <h2>{post.description}</h2>
+        <Comment postID={post.id} />
```
Comment (digi0ps) [inline src/Components/PostsComponent/PostDetails.jsx]: Make component name more meaningful like `CommentsWithInput`

File: src/Components/CommentComponent/Comment.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  commentInputHandler = (e) => {
+    this.setState({ comment: e.target.value })
+  }
+
+  createCommentHandler = async (e) => {
+    const comment = {
+      comment: this.state.comment,
+      createdAt: new Date().getTime(),
+      postID: this.props.postID,
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Why do you need postID in comment structure? It's not being used anywhere.

File: src/Components/CommentComponent/Comment.jsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    this.setState({
+      comment_arr: [newComment, ...this.state.comment_arr],
+      comment: "",
+    })
+  }
+
+  async componentDidMount() {
+    const arr = await getAllComments(this.props.postID)
+    this.setState({
+      comment_arr: [...arr, ...this.state.comment_arr],
```
Comment (digi0ps) [inline src/Components/CommentComponent/Comment.jsx]: Why spreading `...this.state.comment_arr`? On mount it's going to be an empty arr.

**Review by digi0ps — COMMENTED** [review]
Great PR! 🎉 
Have left some minor comments.

Apart from that,
-  group all helpers together or by model (Posts, Comments). Separate file for each function is not needed.
- do you think going to a route based folder structure would be beneficial?

File: src/Components/CommentComponent/Comment.jsx (hunk 1 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Done.

File: src/Components/CommentComponent/Comment.jsx (hunk 3 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Done

File: src/Components/CommentComponent/Comment.jsx (hunk 4 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: map should return something sir (usually an array) ?

Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: ok got your point
`comment.map((comment) => (
              <div key={comment.id}>
                <h3>{comment.comment}</h3>
                <p>{new Date(comment.createdAt).toLocaleTimeString()}</p>
                <cite>by {comment.username}</cite>
              </div>
            ))`

Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Done

File: src/Components/PostsComponent/PostDetails.jsx (hunk 5 above)
Comment (Officer-R) [reply src/Components/PostsComponent/PostDetails.jsx]: Done

File: src/Components/PostsComponent/PostDetails.jsx (hunk 6 above)
Comment (Officer-R) [reply src/Components/PostsComponent/PostDetails.jsx]: Done

File: src/Components/CommentComponent/Comment.jsx (hunk 8 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Done

File: src/Components/PostsComponent/PostDetails.jsx (hunk 7 above)
Comment (Officer-R) [reply src/Components/PostsComponent/PostDetails.jsx]: Done

File: src/Components/CommentComponent/Comment.jsx (hunk 9 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: My bad sir.

Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Thought this was event handler.

File: src/Components/CommentComponent/Comment.jsx (hunk 1 above)
Comment (rohit1101) [reply src/Components/CommentComponent/Comment.jsx:26]: Done

File: src/Components/CommentComponent/Comment.jsx (hunk 2 above)
Comment (Officer-R) [reply src/Components/CommentComponent/Comment.jsx]: Done

**Pull request merged**